### ISSUE TITLE
rustup can't update itself to v1.25.2 on aarch64-pc-windows-msvc platform

### ISSUE THREAD

**Issue by ssg** [rust-lang/rustup#3187]
### Problem

Trying to rustup update from 1.25.1 to 1.25.2 fails to update itself with this error message on ARM64 Windows:

```
error: could not download file from 'https://static.rust-lang.org/rustup/archive/1.25.2/aarch64-pc-windows-msvc/rustup-init.exe' to '..local folder..': http request returned an unsuccessful status code: 404
```

### Steps

1. Run `rustup update`
2. Observe error message

### Possible Solution(s)

Put the missing file in place.

### Rustup version

1.25.1

### Installed toolchains

```console
Default host: aarch64-pc-windows-msvc

stable-aarch64-pc-windows-msvc (default)
rustc 1.67.0 (fc594f156 2023-01-24)
```

**Comment by ehuss** [comment]
Copying my comment from [Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/signature.20failures/near/325302642):

This appears to be an issue with how rustup detects the platform of what it should download.

The installer shell script installs the i686-pc-windows-msvc binary on aarch64. The aarch64 release isn't available on stable, yet (see https://github.com/rust-lang/rustup/issues/2612).

However, the self-update code [here](https://github.com/rust-lang/rustup/blob/d4e1f1708ac4044e5c1a7cd7c032741370221e17/src/dist/dist.rs#L271-L273) detects the native architecture of the system it is running on, and thinks it needs the aarch64 version of rustup.

It's not clear why it is done this way.

A workaround is to force the arch: 

```
RUSTUP_OVERRIDE_HOST_TRIPLE=i686-pc-windows-msvc rustup self update
```

**Comment by arlosi** [comment]
#3010 added the native architecture detection so that rustup would install the native aarch64 Rust toolchain on aarch64 hosts by default (similar to how it installs native x86_64, even though rustup itself can be i686).

I didn't realize it was also used for self-update. The simplest fix would be to close #2612 by enabling the distribution of aarch64 builds of stable rustup. I believe it's using `native-tls` instead of `ring`, but that should be fine.

Feel free to ping me if we need more testing on aarch64 hw. I have a Surface Pro X.

**Comment by Alovchin91** [comment]
Hi @arlosi @ehuss  I’ve been bumping this thread for half a year now but nobody ever responded: https://github.com/rust-lang/rustup/issues/2612#issuecomment-1219979692

Ring is hardly happening, and WebPKI is the next unmaintained project to update.

Why can’t we just release the native-tls version of rustup-init on Windows arm64 for now (the one that is currently built on master)? We can always switch later if rustls is ever unblocked.

**Comment by ssg** [comment]
@Alovchin91 I see that the last update to ring was 2 months ago. It looks far from abandoned. But, I agree that it shouldn't be a blocker for rustup.

**Comment by Alovchin91** [comment]
@ssg I didn't mean Ring is abandoned, but it's clearly not a priority for the author at the moment 🤷‍♂️ I've [contributed Windows ARM support](https://github.com/briansmith/ring/pull/1339) to ring-rs almost 1,5 years ago; since then there was no single release.

**Comment by rbtcollins** [comment]
I'm going to close this bug in favor of #2612.
